pymodbus/transport/transport.py

```diff
@@ -156,17 +156,21 @@ def __init__(
         self.is_server = is_server
         self.is_closing = False
 
-        self.transport: asyncio.BaseTransport = None
-        self.loop: asyncio.AbstractEventLoop = None
+        self.loop: asyncio.AbstractEventLoop = None  # type: ignore[assignment]
+        self.transport: asyncio.BaseTransport | None = None
```

Comment:
And here you use it....any maintainer will think why on earth is self.loop without None and this one is with None...both are later used in the same way.

Reply:
The later callers are different.  Calling code checks `self.transport != None` in all places.  But the `loop()` code is more complicated.